### PR TITLE
Serialized reporter fixes

### PR DESCRIPTION
**Pull request by mcimbora**
Depends on https://github.com/radargun/radargun/pull/253.

Fixes problem of ReporterHelper initialization - static initializer threw exception as it attempted to register the same reporers multiple times. As bin/report.sh adds contents of reporter-default on classpath, wrong plugin.properties file was read when registering reporter-perfrepo (the one included in reporter-default).

### CODE REVIEW

**Comment by mcimbora** [comment]
Added option to specify multiple result directories.

**Comment by alanfx** [comment]
Hey @mcimbora, do you want these changes on master, branch_2.x, or both? I guess we should figure out a process for that...

**Comment by mcimbora** [comment]
@alanfx I agree. In my view it would be best to create 2 separate PRs, each of them having its specific target milestone (2.2.0/3.0.0). This makes reviewer's life a bit easier as some commits require further changes before integrating (e.g. make sure right version is used when new pom.xml is added, fixing merge conflicts, etc.). This one should go to both branches. @alanfx wdyt?

**Comment by rvansa** [comment]
@mcimbora @alanfx There should be 2 PRs, but if the PR is non-trivial, PR for master should come first and for 2.x branch only after the reviewer is confident that there'll be no more modifications needed.

**Comment by alanfx** [comment]
@mcimbora I agree with the process described by @rvansa. The handle_pull_request.py will need to be modified to handle multiple branches, though. Unless I am the only one still using this script!

File: core/src/main/java/org/radargun/reporting/ReporterHelper.java [hunk 1]
```diff
@@ -83,7 +85,8 @@ public static void loadReporters(File reporterDir, Map<String, Class<? extends R
                } else {
                   log.debug("Registering reporter '" + type + "' using " + clazzName);
                   try {
-                     Class<? extends Reporter> reporterClazz = (Class<? extends Reporter>) classLoader.loadClass(clazzName);
+                     ClassLoader reporterHelperClassLoader = new URLClassLoader(urls.toArray(new URL[urls.size()]), ReporterHelper.class.getClassLoader());
```
Comment (rvansa) [inline core/src/main/java/org/radargun/reporting/ReporterHelper.java]: Why do you create another ClassLoader and don't use the reporterSpecificClassloader?

File: plugins/infinispan52/src/main/java/org/radargun/service/AbstractConfigurationProvider.java [hunk 2]
```diff
@@ -36,14 +36,17 @@
 
    private void loadConfigFile(String filename, Map<String, byte[]> configs) {
       if (filename != null) {
+         InputStream is = null;
          try {
-            InputStream is = getClass().getClassLoader().getResourceAsStream(filename);
+            is = getClass().getClassLoader().getResourceAsStream(filename);
```
Comment (rvansa) [inline plugins/infinispan52/src/main/java/org/radargun/service/AbstractConfigurationProvider.java]: Any reason to not use try-with-resources here?

**Comment by rvansa** [comment]
It seems to me that adding the reporters directly on classpath in reporter.sh is a mistake. We should rather add id to REPORTER_PATHS as the default value.

**Comment by rvansa** [comment]
@rmacor This PR is stalled here for a while, will you react?

**Comment by rvansa** [comment]
@rmacor oops, sorry, this was meant for @mcimbora

**Comment by mcimbora** [comment]
@rvansa Sorry, other priorities ATM. We can close this & I'll reopen when ready if that makes sense to you.

**Comment by rvansa** [comment]
@mcimbora Since it's not a new feature but bugfix, I'd prefer to keep it open. But it would be best if you found some time to take a look sometime in next few weeks.

**Comment by mcimbora** [comment]
Closing, fixed by https://github.com/radargun/radargun/commit/2931c23a17212a57223978b970bb8653f1928bc4

**Pull request closed without merging**